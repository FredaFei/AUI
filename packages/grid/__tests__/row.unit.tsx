import * as renderer from 'react-test-renderer'
import React from 'react'
import {shallow, mount} from 'enzyme'

import Col from '../col'
import Row from "../row";

describe('测试 Row 以下行为', () => {
  it('Row 存在', () => {
    const json = renderer.create(<Row>Row-12</Row>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('Row 设置 align,gutter,verticalAlign 默认值分别为left，0，top', () => {
    const wrapper = mount(
      <Row>
        <Col span={12}>col-12</Col>
        <Col span={12}>col-12</Col>
      </Row>)
    expect(wrapper.props().align).toEqual('left')
    expect(wrapper.props().gutter).toEqual(0)
    expect(wrapper.props().verticalAlign).toEqual('top')
  })
  it('Row 设置 align', () => {
    const wrapper = mount(
      <Row align="right" verticalAlign="center">
        <Col span={12}>
          <div style={{height: '160px'}}>col-12-1</div>
        </Col>
        <Col span={4}>
          <div style={{height: '80px'}}>col-4-1</div>
        </Col>
      </Row>)
    expect(wrapper.props().align).toEqual('right')
    expect(wrapper.find('.am-row').hasClass('am-row-align-right')).toEqual(true)
  })
  it('Row 设置 verticalAlign', () => {
    const wrapper = mount(
      <Row align="right" verticalAlign="center">
        <Col span={12}>
          <div style={{height: '160px'}}>col-12-1</div>
        </Col>
        <Col span={4}>
          <div style={{height: '80px'}}>col-4-1</div>
        </Col>
      </Row>)
    expect(wrapper.props().verticalAlign).toEqual('center')
    expect(wrapper.find('.am-row').hasClass('am-row-verticalAlign-center')).toEqual(true)
  })
  it('Row 设置 gutter', () => {
    const wrapper = mount(
      <Row gutter={10}>
        <Col span={6}>
          <div>col-6</div>
        </Col>
        <Col span={6}>
          <div>col-6</div>
        </Col>
        <Col span={6}>
          <div>col-6</div>
        </Col>
        <Col span={6}>
          <div>col-6</div>
        </Col>
      </Row>)
    expect(wrapper.props().gutter).toEqual(10)
    expect(wrapper.find('.am-col-gutter').length).toBe(4)
  })

  it('Row 设置 自定义类名', () => {
    const wrapper = shallow(<Row className="test">Row-12</Row>)
    expect(wrapper.find('.am-row').hasClass('test')).toBe(true)
  })
  it('Row 设置 自定义样式', () => {
    const wrapper = shallow(<Row style={{color: 'red'}}>Row-12</Row>)
    expect(wrapper.render()).toMatchSnapshot()
  });
})