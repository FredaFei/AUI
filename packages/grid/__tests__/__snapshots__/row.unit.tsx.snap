// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Row 以下行为 Row 存在 1`] = `
<div
  className="am-row "
  data-role="Row"
  style={
    Object {
      "marginLeft": "0px",
      "marginRight": "0px",
    }
  }
>
  Row-12
</div>
`;

exports[`测试 Row 以下行为 Row 设置 自定义样式 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "am-row ",
      "data-role": "Row",
      "style": "margin-left:0px;margin-right:0px;color:red",
    },
    "children": Array [
      Object {
        "data": "Row-12",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-role": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-role": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
